pkg/osquery/tables/xfconf/xfconf.go

```diff
@@ -0,0 +1,164 @@
+//go:build !windows
+// +build !windows
+
+package xfconf
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/launcher/pkg/dataflatten"
+	"github.com/kolide/launcher/pkg/osquery/tables/dataflattentable"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+// Provides configuration settings for devices using XFCE desktop environment.
+// See https://docs.xfce.org/xfce/xfce4-session/advanced#files_and_environment_variables
+// for a list of the files and environment variables we check for configuration.
+
+var xfconfChannelXmlPath string = filepath.Join("xfce4", "xfconf", "xfce-perchannel-xml")
+
+type XfconfQuerier struct {
+	logger log.Logger
+}
+
+func TablePlugin(logger log.Logger) *table.Plugin {
+	t := &XfconfQuerier{
+		logger: logger,
+	}
+
+	return table.NewPlugin("kolide_xfconf", dataflattentable.Columns(table.TextColumn("username"), table.TextColumn("channel")), t.generate)
+}
+
+func (t *XfconfQuerier) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+
+	users := tablehelpers.GetConstraints(queryContext, "username")
+	if len(users) < 1 {
+		return results, errors.New("kolide_xfconf requires at least one username to be specified")
+	}
+
+	for _, username := range users {
+		u, err := user.Lookup(username)
+		if err != nil {
+			return nil, fmt.Errorf("finding user by username '%s': %w", username, err)
+		}
```

Comment:
I'm not sure if this should be a hard error (as you have it) or a soft error (eg: continue with a log message). Hard errors will cause a large blob of SQL o fail, soft errors are more graceful.

For something like the user check, hard error makes sense. It's a syntactic failure. This feels a bit closer to an exec or OS failure, and we probably don't want to through an error. 

But I could go either way

Reply:
I can switch to a soft error -- I see e.g. gsettings does a soft error in a similar situation.

Reply:
We used to use a lot of hard errors, then we had issues, now we're moving to a lot more soft errors. It's not yet clear to me what ideal is